addon/lib/helperworker.js

```diff
@@ -31,6 +50,64 @@ function resetPageMod(backend) {
     contentScriptFile: [self.data.url("viewerworker.js")],
     onAttach: function (worker) {
       watchWorker(worker);
+
+      worker.port.on("requestAccount", watchFunction(function (info) {
+        let { id, options: { action } } = info;
+
+        let currentProfile = user.getProfileInfo();
+        if (currentProfile && currentProfile.email) {
+          let err = errors.extAlreadySignedIn();
+          worker.port.emit("account", id, toErrorResponse(err));
+          return;
+        }
+
+        if (action != "signup" && action != "signin") {
+          let err = errors.badParams();
+          worker.port.emit("account", id, toErrorResponse(err));
+          return;
+        }
+
+        let handler = new user.OAuthHandler(backend);
+        handler.getOAuthParams().then(defaults => {
+          worker.port.emit("account", id, toSuccessResponse());
+
+          let params = Object.assign({ action }, defaults);
+          return handler.logInWithParams(params).then(response => {
+            return handler.getProfileInfo();
+          }).then(profile => {
+            // TODO: Don't clobber profile info if the user has already set a
```

Comment:
Maybe a separate `setDefaultProfileInfo` call that only updates your nickname and avatar if you haven't set one?